pkg/tenant/members.go

```diff
@@ -238,66 +242,94 @@ func (s *Server) MemberUpdate(c *gin.Context) {
 	var memberID ulid.ULID
 	if memberID, err = ulid.Parse(c.Param("memberID")); err != nil {
 		sentry.Warn(c).Err(err).Str("id", c.Param("memberID")).Msg("could not parse member id")
-		c.JSON(http.StatusNotFound, api.ErrorResponse("member not found"))
+		c.JSON(http.StatusNotFound, api.ErrorResponse(responses.ErrMemberNotFound))
 		return
 	}
 
 	// Bind the user request with JSON and return a 400 response
 	// if binding is not successful.
-	if err = c.BindJSON(&member); err != nil {
+	if err = c.BindJSON(&req); err != nil {
 		sentry.Warn(c).Err(err).Msg("could not parse member update request")
-		c.JSON(http.StatusBadRequest, api.ErrorResponse(api.ErrUnparsable))
+		c.JSON(http.StatusBadRequest, api.ErrorResponse(responses.ErrTryLoginAgain))
 		return
 	}
 
-	// Verify the member email exists.
-	if member.Email == "" {
-		c.JSON(http.StatusBadRequest, api.ErrorResponse("member email is required"))
-		return
-	}
+	var member *db.Member
+	if member, err = db.RetrieveMember(c.Request.Context(), orgID, memberID); err != nil {
+		if errors.Is(err, db.ErrNotFound) {
+			c.JSON(http.StatusNotFound, api.ErrorResponse(responses.ErrMemberNotFound))
+			return
+		}
 
-	// Verify the member role exists and return a 400 response if it doesn't.
-	if member.Role == "" {
-		c.JSON(http.StatusBadRequest, api.ErrorResponse("member role is required"))
+		sentry.Error(c).Err(err).Msg("could not retrieve member from the database")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse(responses.ErrSomethingWentWrong))
 		return
 	}
 
-	// Verify the role provided is valid.
-	if !perms.IsRole(member.Role) {
-		c.JSON(http.StatusBadRequest, api.ErrorResponse("unknown member role"))
+	// Update all fields provided by the user
+	req.Normalize()
+	member.Name = req.Name
+	member.ProfessionSegment = req.ProfessionSegment
+	member.DeveloperSegment = req.DeveloperSegment
+
+	if !member.Invited {
+		member.Organization = req.Organization
+		member.Workspace = req.Workspace
+	}
+
+	// Validate the member update
+	if err = member.Validate(); err != nil {
```

Comment:
Okay this is a bit awkward because we're validating the database record twice (here and in `db.UpdateMember`), but I wanted to make sure that the organization and workspace are valid before sending it to Quarterdeck.

Reply:
I'm not bothered by double validation - but I would recommend that you add that note to the comment so we remember why we're doing it twice in the future.